README.md

```diff
@@ -1 +1,138 @@
 # web3-file
+
+The [Web File](https://developer.mozilla.org/en-US/docs/Web/API/File) implementation for Web3.
+
+## Motivation
+
+The [Web File](https://developer.mozilla.org/en-US/docs/Web/API/File) implementation and standard are still on their beginning.
+
+The current implementation has a non standard [**File.webkitRelativePath**](https://developer.mozilla.org/en-US/docs/Web/API/File/webkitRelativePath) `read-only` property that we cannot use outside the `webkitdirectory` context. As a result, we cannot set a File path programatically in the Browser, nor in Node.js [@web-std/file](https://github.com/web-std/io/tree/main/file).
+
+`Web3File` bridges this gap by adding a path property and other extras, while being a close implementation of Web File. This allows 
+files to be transformed into their [UnixFs](https://github.com/ipfs/specs/blob/master/UNIXFS.md) representation and back into the original structure, and leverage the IPFS Content Routing primitives.
+
+## Install
+
+```sh
+# install it as a dependency
+$ npm i web3-file
+```
+
+## API
+
+To create your `Web3File` with an async iterable content, you can simply create an instance of Web3File. Otherwise, you can leverage one of the static functions provided to create a `Web3File` from another data type.
+
+Please note that the same options of the constructor can be provided in the static functions.
+
+### class Web3File
+
+| Name | Type | Description |
+|------|------|-------------|
+| filename | `string` | filename |
+| content | `AsyncIterable<Uint8Array>` | File content to be read |
+| [options] | `object` | Web3File options |
+| [options.path] | `string` | File Path |
+| [options.lastModified] | `number` | Last modified timestamp |
+
+```js
+import Web3File from 'web3-file'
+
+const file = new Web3File(
+  'file.zip',
+  fs.createReadStream('path/to/file.zip')
```

Comment:
We should flip these around so that data is first and filename is second so that it is more like the `File` constructor.

Same for the statc factory functions.

```suggestion
  fs.createReadStream('path/to/file.zip'),
  'file.zip'
```

Reply:
yes, I agree! I will shift them everywhere